core/src/main/java/org/apache/iceberg/MetadataUpdateParser.java

```diff
@@ -383,7 +383,7 @@ private static MetadataUpdate readUpgradeFormatVersion(JsonNode node) {
 
   private static MetadataUpdate readAddSchema(JsonNode node) {
     Preconditions.checkArgument(node.hasNonNull(SCHEMA), "Cannot parse missing field: schema");
-    JsonNode schemaNode = node.get(SCHEMA);
+    JsonNode schemaNode = JsonUtil.get(SCHEMA, node);
```

Comment:
Consider relying on the `Precondition` inside of `JsonUtil` (and have it use `hasNonNull`). That pattern seems to exist a lot still and could be pushed into `JsonUtil`.

Reply:
makes total sense, updated